src/swaps/liqualityboost/lbspERC20toNative/LbspERC20toNative.js

```diff
@@ -0,0 +1,345 @@
+        toAccountId: quote.fromAccountId
+      },
+      feePrices,
+      max
+    })
+
+    // 'from' asset -> bridge asset
+    if (!isERC20(asset) && txType === LbspERC20toNative.txTypes.SWAP) {
+      const automatedMarketMakerFees = await this.bridgeAssetToAutomatedMarketMaker[
+        quote.bridgeAsset
+      ].estimateFees({
+        network,
+        walletId,
+        asset,
+        txType: LbspERC20toNative.txTypes.SWAP,
+        quote: {
+          ...quote,
+          to: quote.bridgeAsset,
+          toAmount: quote.bridgeAssetAmount,
+          slippagePercentage
+        },
+        feePrices,
+        max
+      })
+      const totalFees = {}
+      for (const key in automatedMarketMakerFees) {
+        totalFees[key] = BN(automatedMarketMakerFees[key]).plus(liqualityFees[key])
+      }
+
+      return totalFees
+    }
+
+    return liqualityFees
+  }
+
+  async finalizeAutomatedMarketMakerAndStartLiqualitySwap({
+    swapLiqualityFormat,
+    swapAMMFormat,
+    network,
+    walletId
+  }) {
+    let result = await this.bridgeAssetToAutomatedMarketMaker[
+      swapAMMFormat.bridgeAsset
+    ].waitForSwapConfirmations({
+      swap: swapAMMFormat,
+      network,
+      walletId
+    })
+
+    if (result?.status === 'SUCCESS') {
+      result = await this.liqualitySwapProvider.newSwap({
+        network,
+        walletId,
+        quote: swapLiqualityFormat
+      })
+
+      return {
+        ...result,
+        ...swapLiqualityFormat,
+        toAmount: result.toAmount,
+        status: result.status
+      }
+    }
+  }
+
+  async performNextSwapAction(store, { network, walletId, swap }) {
+    let updates
+    const swapLiqualityFormat = {
+      ...swap,
+      from: swap.bridgeAsset,
+      fromAmount: swap.bridgeAssetAmount,
+      toAccountId: swap.fromAccountId,
+      slippagePercentage
+    }
+
+    const swapAutomatedMarketMakerFormat = {
+      ...swap,
+      to: swap.bridgeAsset,
+      toAmount: swap.bridgeAssetAmount,
+      slippagePercentage
+    }
+
+    if (swap.status === 'WAITING_FOR_SWAP_CONFIRMATIONS') {
+      updates = await withInterval(async () =>
+        this.finalizeAutomatedMarketMakerAndStartLiqualitySwap({
+          swapLiqualityFormat,
+          swapAMMFormat: swapAutomatedMarketMakerFormat,
+          network,
+          walletId
+        })
+      )
+    } else {
+      updates = await this.liqualitySwapProvider.performNextSwapAction(store, {
+        network,
+        walletId,
+        swap: swapLiqualityFormat
+      })
+    }
+
+    if (!updates && !LbspERC20toNative.lspEndStates.includes(swap.status)) {
+      updates = await this.bridgeAssetToAutomatedMarketMaker[
+        swap.bridgeAsset
+      ].performNextSwapAction(store, {
+        network,
+        walletId,
+        swap: swapAutomatedMarketMakerFormat
+      })
+    }
+
+    if (!updates) return
+
+    return {
+      ...updates,
+      // reset from and to assets and values
+      from: swap.from,
```

Comment:
If you reset the from value isn't this resulting in showing wrong information in the timeline view. At the worst case we should keep the from value in some separate property I believe and handle it in the timeline view. 

E.g when you start a swap from sovryn -> eth , if you come to this step where you already exchanged sovryn to rbtc it will update the from value and in timeline diagram you will see that your swap is from rbtc -> eth, but user is swapping sov -> eth. I am not sure if I am correct but you should check this in the timeline view

Reply:
No. It's quite the opposite. All 'initial' data is kept in 'swap' objet completely intact. In the start of `performNextSwapAction` method the two objects `swapLiqualityFormat` and  `swapAutomatedMarketMakerFormat` format the data so it will be suitable for the AMM or LSP. At the end of the method data is again formatted in a way that could be correctly shown in time line.